frame/wormhole/backing/s2s/src/lib.rs

```diff
@@ -168,8 +175,13 @@ pub mod pallet {
 	/// `(sender, amount)` the user *sender* lock and remote issuing amount of asset
 	#[pallet::storage]
 	#[pallet::getter(fn transaction_infos)]
-	pub type TransactionInfos<T: Config> =
-		StorageMap<_, Identity, u64, (AccountId<T>, RingBalance<T>), ValueQuery>;
+	pub type TransactionInfos<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		BridgeMessageId,
+		(AccountId<T>, RingBalance<T>),
+		ValueQuery,
+	>;
```

Comment:
Might need to review and check wether we need migration this.

Since it only affects testnets, we can also take simple operations by checking there is no ongoing transaction infos & simply remove and replace.

Reply:
Yes we need migration here for production environment. I want to simply replace it for testnet in this upgrade, and also replace the mapping-token contract directly to test the registration process.

Reply:
But to avoid strange issues in testnets in future, I would still do clean for the testnet runtimes (for example, remove old storage in runtime upgrade).

And for there, suggest to use OptionQuery instead, then we can use `if Some(...) = ` to replace existing ambiguity way:

```
let (user, amount) = <TransactionInfos<T>>::take(message_id);
				if amount.is_zero() {
					continue;
				}
```